UI/include/Hierarchy.h

```diff
@@ -7,11 +7,11 @@
 #include <vector>
 #include <memory>  // Use smart pointers for safe memory management
 #include "../GameObject.h"
-//#include "Entity.h"
+#include "Entity.h"
+#include "Scene.h"
 
-void drawHierarchyLine(GameObject* obj);
-void drawHierarchy();
-GameObject* getSelectedGameObject();
-std::string getNextRootObjectName();
-extern std::vector<GameObject*> gameObjects;
+void drawHierarchyLine(Entity* entity);
+void drawHierarchy(Scene* s);
```

Comment:
same thing as before, abbreviated formal parameter (i'm not entirely sure how picky we want to be on this as a whole, just pointing out in case)